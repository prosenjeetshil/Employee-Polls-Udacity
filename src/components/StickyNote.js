import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { formatDate } from "../utils/helpers";
import { Card, Avatar, Button, Typography } from "antd";

const { Text } = Typography;

const StickyNote = ({ id, name, avatarURL, timestamp }) => {
  return (
    <Card className="sticky-note">
      <div className="sticky-note__header">
        <Avatar size={48} src={avatarURL} />
        <div className="sticky-note__header-text">
          <Text strong>{name}</Text>
          <Text type="secondary">{formatDate(timestamp)}</Text>
        </div>
      </div>
      <Link to={`/questions/${id}`}>
        <Button type="primary" block>
          Show
        </Button>
      </Link>
    </Card>
  );
};

const mapStateToProps = ({ users, questions }, { id }) => {
  const question = questions[id];

  return {
    id,
    name: users[question.author].name,
    avatarURL: users[question.author].avatarURL,
    timestamp: question.timestamp,
  };
};

export default connect(mapStateToProps)(StickyNote);
