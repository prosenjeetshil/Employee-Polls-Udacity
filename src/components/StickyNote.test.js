import React from "react";
import { shallow } from "enzyme";
import { Link } from "react-router-dom";
import StickyNote from "./StickyNote";

describe("StickyNote component", () => {
  const props = {
    id: "123",
    name: "Jane Doe",
    avatarURL: "www.example.com/avatar.jpg",
    timestamp: 1606952400000,
  };
  const wrapper = shallow(<StickyNote {...props} />);

  it("renders without crashing", () => {
    expect(wrapper).toBeDefined();
  });

  it("renders an Avatar component", () => {
    expect(wrapper.find("Avatar").length).toBe(1);
  });

  it("renders a Link component", () => {
    expect(wrapper.find(Link).length).toBe(1);
  });

  it("renders a Text component", () => {
    expect(wrapper.find("Text").length).toBe(2);
  });

  it("renders a Button component", () => {
    expect(wrapper.find("Button").length).toBe(1);
  });

  it("renders the name of the author", () => {
    expect(wrapper.find("Text").at(0).prop("children")).toBe(props.name);
  });

  it("renders the formatted timestamp", () => {
    expect(wrapper.find("Text").at(1).prop("children")).toBe("Oct 15, 2021");
  });

  it("renders the correct URL in the Link component", () => {
    expect(wrapper.find(Link).prop("to")).toBe(`/questions/${props.id}`);
  });
});
