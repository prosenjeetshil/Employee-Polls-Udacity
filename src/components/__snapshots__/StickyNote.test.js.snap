// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickyNote should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="column is-half-desktop is-one-third-widescreen is-one-quarter-fullhd"
      >
        <div
          class="card"
        >
          <div
            class="card-content"
          >
            <div
              class="media"
            >
              <div
                class="media-left"
              >
                <figure
                  class="image is-48x48"
                >
                  <img
                    alt="Sarah Edo"
                    src="https://placekitten.com/g/240/240"
                  />
                </figure>
              </div>
              <div
                class="media-content"
              >
                <p
                  class="title is-6"
                >
                  Sarah Edo
                </p>
                <p
                  class="subtitle is-7"
                >
                  9:21:PM | 6/28/2016
                </p>
              </div>
            </div>
            <a
              class="button is-primary is-rounded is-fullwidth"
              data-testid="show-button"
              href="/questions/8xf0y6ziyjabvozdd253nd"
            >
              Show
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="column is-half-desktop is-one-third-widescreen is-one-quarter-fullhd"
    >
      <div
        class="card"
      >
        <div
          class="card-content"
        >
          <div
            class="media"
          >
            <div
              class="media-left"
            >
              <figure
                class="image is-48x48"
              >
                <img
                  alt="Sarah Edo"
                  src="https://placekitten.com/g/240/240"
                />
              </figure>
            </div>
            <div
              class="media-content"
            >
              <p
                class="title is-6"
              >
                Sarah Edo
              </p>
              <p
                class="subtitle is-7"
              >
                9:21:PM | 6/28/2016
              </p>
            </div>
          </div>
          <a
            class="button is-primary is-rounded is-fullwidth"
            data-testid="show-button"
            href="/questions/8xf0y6ziyjabvozdd253nd"
          >
            Show
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
